xdsl/dialects/builtin.py

```diff
@@ -350,18 +350,22 @@ class DictionaryAttr(GenericData[dict[str, _DictionaryAttrT]]):
     name = "dictionary"
 
     @staticmethod
-    def parse_parameter(parser: Parser) -> dict[_DictionaryAttrT]:
+    def parse_parameter(
+        parser: Parser,
+        parse_optional_attribute: Callable[[], _DictionaryAttrT | None]
+    ) -> dict[str, _DictionaryAttrT]:
         parser.parse_char("{")
-        data = parser.parse_dictionary(parser.parse_optional_attribute)
+        data = parser.parse_dictionary(parser.parse_str_literal,
```

Comment:
Shouldn't this use `parse_optional_str_literal`?

Reply:
It didn't come up in my autocomplete!